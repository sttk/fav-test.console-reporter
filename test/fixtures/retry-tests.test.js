'use strict';

var Promise = (typeof Promise === 'function') ? Promise :
              require('promise-polyfill');
var chkEnv = require('../tool/chk-env');

var chai = require('chai');
if (chkEnv.isSupportES6()) {
  var chaiAsPromised = require('chai-as-promised');
  chai.use(chaiAsPromised);
}
var expect = chai.expect;

var browser = {
  get: function() {},
};
var count = 0;
var $ = function() {
  return {
    isDisplayed: function() {
      count++;
      return Promise.resolve(count > 3);
    },
  };
};

describe('retries (2 times)', function() {
  // Retry all tests in this suite up to 4 times
  this.retries(4);

  before(function() {
    count = 0;
  });

  beforeEach(function() {
    browser.get('http://www.yahoo.com');
  });

  it('should succeed on the 3rd try', function() {
    // Specify this test to only retry up to 2 times
    this.retries(2);
    if (chkEnv.isSupportES6()) {
      return expect($('.foo').isDisplayed()).to.eventually.be.true;
    } else {
      return $('.foo').isDisplayed().then(function(value) {
        expect(value).to.be.true;
      });
    }
  });
});

describe('retries (3 times)', function() {
  // Retry all tests in this suite up to 4 times
  this.retries(4);

  before(function() {
    count = 0;
  });

  beforeEach(function() {
    browser.get('http://www.yahoo.com');
  });

  it('should succeed on the 3rd try', function() {
    this.retries(3);
    if (chkEnv.isSupportES6()) {
      return expect($('.foo').isDisplayed()).to.eventually.be.true;
    } else {
      return $('.foo').isDisplayed().then(function(value) {
        expect(value).to.be.true;
      });
    }
  });
});

