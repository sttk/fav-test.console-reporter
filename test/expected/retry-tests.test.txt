

  retries (2 times)
    1) should succeed on the 3rd try

  retries (3 times)
    âœ“ should succeed on the 3rd try


  1 passing (13ms)
  1 failing

  1) retries (2 times)
       should succeed on the 3rd try:

      AssertionError: expected false to be true
      + expected - actual

      -false
      +true





